packages/medusa/src/api/routes/store/products/index.ts

```diff
@@ -9,10 +9,15 @@ import { validateProductSalesChannelAssociation } from "../../../middlewares/pub
 import { validateSalesChannelParam } from "../../../middlewares/publishable-api-key/validate-sales-channel-param"
 import { StoreGetProductsParams } from "./list-products"
 import { StoreGetProductsProductParams } from "./get-product"
+import { FlagRouter } from "../../../../utils/flag-router"
 
 const route = Router()
 
-export default (app) => {
+export default (app, featureFlagRouter: FlagRouter) => {
+  if (featureFlagRouter.isFeatureEnabled("product_categories")) {
+    allowedStoreProductsRelations.push("categories")
```

Comment:
question: Should we add it to the default relations or do we expect the user to use the expand for that?

Reply:
Since its adding a separate join on an already performance heavy endpoint, this should be optional or intentional.